drafts/npm.md

```diff
@@ -0,0 +1,472 @@
+   the npm registry performs scans to detect typosquatting, no system is
+   perfect, so stay vigilant. Identify the GitHub repository of the package and
+   assess its trustworthiness through, for example, its number of contributors,
+   stars, etc.
+
+    1. Note: Non-ASCII characters and uppercase ASCII characters are [not
+       supported in npm package
+       names](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#name),
+       so developers need not worry about homograph attacks, whereby an attacker
+       names their package using non-ascii characters that render similar to an
+       ascii character.
+
+2. When you identify a GitHub project of interest, follow their documentation
+   to identify the corresponding package name.
+
+   Use [OpenSSF Security Scorecards](http://github.com/ossf/scorecard) to
+   learn about the current security posture of the dependency.
+
+3. Use [deps.dev](http://deps.dev) to learn about the security posture of
+   transitive dependencies using. You may list transitive dependencies by using
+   [component analysis](https://github.com/microsoft/component-detection/).
+
+4. Use [npm-audit](https://docs.npmjs.com/cli/v8/commands/npm-audit) to learn
+   about existing vulnerabilities in the dependencies of the project.
+
+**Warning**: Organization verification does not exist on the npm registry and
+it's "first-come first serve." It is possible to challenge the owner of an org
+for squatting after the fact using the [dispute
+process](https://docs.npmjs.com/policies/disputes#beginning-the-process).
+
+### Declaration
+
+In npm, the
+[package.json](https://docs.npmjs.com/cli/v6/configuring-npm/package-json)
+describes a project (name, components, dependencies, etc.). Dependencies may be
+defined by a package name, url, repo, etc. Additional constraints may be
+defined for each dependency, such as which tags, branches, versions or
+commitish are allowed.
+
+**Note**: The manifest file ***does not*** list transitive dependencies; it
+lists only direct project dependencies.
+
+### Reproducible installation
+
+A reproducible installation is one that guarantees the exact same copy the
+dependencies are used each time the package is installed. This has various
+benefits, including:
+
+- Ensuring the dependencies installed are the ones declared and reviewed via
+  pull requests.
+
+- Helping quickly identify possible compromises of your infrastructure if one
+  of your dependencies is found to have vulnerabilities, as you will be able to
+  quickly determine the commit range when your repository was at risk.
+
+- Mitigating certain threats such as malicious dependencies. A recent example
+  is the [color package's
+  sabotage](https://blog.sonatype.com/npm-libraries-colors-and-faker-sabotaged-in-protest-by-their-maintainer-what-to-do-now):
+  without a reproducible installation, developers installed and ran a newly
```

Comment:
this seems disingenuous - it wouldn't have mattered if it was reproducible, because the developer intentionally sabotaged the project.